rts/Lua/LuaSyncedRead.cpp

```diff
@@ -3632,14 +3606,12 @@ int LuaSyncedRead::GetProjectilesInRectangle(lua_State* L)
 				if (pro->piece && excludePieceProjectiles)
 					continue;
 
-				lua_pushnumber(L, pro->id);
-				lua_rawseti(L, -2, arrayIndex++);
+				lua_pushinteger(L, static_cast<lua_Number>(pro->id));
+				lua_rawseti(L, -2, static_cast<int>(arrayIndex++));
```

Comment:
Casting `pro->id` should no longer be needed. For `arrayIndex` perhaps it would be good to change that var to be int and also remove the cast.

Reply:
Done. Thx.